src/EnergyPlus/CMakeLists.txt

```diff
@@ -738,11 +738,18 @@ if ((CMAKE_COMPILER_IS_GNUCXX AND CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.1) O
     ( ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" OR "${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang") AND (CMAKE_CXX_COMPILER_VERSION VERSION_LESS 9.0)))
   target_link_libraries(energyplusparser PRIVATE stdc++fs)
 endif()
+if(ENABLE_UNITY)
+  set_target_properties(energyplusparser PROPERTIES UNITY_BUILD ON)
+endif()
 
 # first we will create a static library of EnergyPlus this will be linked
 # statically to create the DLL and also the unit tests
 add_library(energypluslib STATIC ${SRC})
 
+if(ENABLE_UNITY)
+  set_target_properties(energypluslib PROPERTIES UNITY_BUILD ON)
+endif()
```

Comment:
This is the only location where I have a question. How does the UNITY_BUILD play out with the precompiled headers? Cf line 773 below, where we link to the `cpp_pch_files` library, which is enabled by default and defined at 

https://github.com/NREL/EnergyPlus/blob/a844dbb1ab3424f6808d460cf638acb04d455ce1/CMakeLists.txt#L29-L45

Reply:
Both options work fine together and their build time reduction effects will cumulate (although with serious diminishing returns, since with unity build there are less translation units to compile).

There is also no point in activating `UNITY_BUILD` for the `cpp_pch_files` target itself if this is what you are asking: it has a single generated `cmake_pch.hxx.cxx` file which cannot really be compiled with any other, and its precompiled results will be force-inlined in all translation units of the targets that depend on `cpp_pch_files` - so in the case of unity build, the auto-generated `unity_XX_cxx.cxx` files.

Reply:
Got it.